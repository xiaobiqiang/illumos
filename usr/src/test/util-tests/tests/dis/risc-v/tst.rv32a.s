/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test our disassembly of the RV32A instructions. Instructions are ordered per the
 * ISA manual.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	lr.w		s0, (s1)
	lr.w.aq		s1, (s2)
	lr.w.rl		s2, (s3)
	lr.w.aqrl	s3, (s4)
	sc.w		s0, s2, (s1)
	sc.w.aq		s1, s3, (s1)
	sc.w.rl		s2, s4, (s1)
	sc.w.aqrl	s3, s5, (s1)
	amoswap.w	t0, t1, (t2)
	amoswap.w.aq	t1, t2, (t3)
	amoswap.w.rl	t2, t3, (t4)
	amoswap.w.aqrl	t3, t4, (t5)
	amoadd.w	t0, t1, (t2)
	amoadd.w.aq	t1, t2, (t3)
	amoadd.w.rl	t2, t3, (t4)
	amoadd.w.aqrl	t3, t4, (t5)
	amoxor.w	t0, t1, (t2)
	amoxor.w.aq	t1, t2, (t3)
	amoxor.w.rl	t2, t3, (t4)
	amoxor.w.aqrl	t3, t4, (t5)
	amoand.w	t0, t1, (t2)
	amoand.w.aq	t1, t2, (t3)
	amoand.w.rl	t2, t3, (t4)
	amoand.w.aqrl	t3, t4, (t5)
	amoor.w		t0, t1, (t2)
	amoor.w.aq	t1, t2, (t3)
	amoor.w.rl	t2, t3, (t4)
	amoor.w.aqrl	t3, t4, (t5)
	amomin.w	t0, t1, (t2)
	amomin.w.aq	t1, t2, (t3)
	amomin.w.rl	t2, t3, (t4)
	amomin.w.aqrl	t3, t4, (t5)
	amomax.w	t0, t1, (t2)
	amomax.w.aq	t1, t2, (t3)
	amomax.w.rl	t2, t3, (t4)
	amomax.w.aqrl	t3, t4, (t5)
	amominu.w	t0, t1, (t2)
	amominu.w.aq	t1, t2, (t3)
	amominu.w.rl	t2, t3, (t4)
	amominu.w.aqrl	t3, t4, (t5)
	amomaxu.w	t0, t1, (t2)
	amomaxu.w.aq	t1, t2, (t3)
	amomaxu.w.rl	t2, t3, (t4)
	amomaxu.w.aqrl	t3, t4, (t5)
.size libdis_test, [.-libdis_test]
