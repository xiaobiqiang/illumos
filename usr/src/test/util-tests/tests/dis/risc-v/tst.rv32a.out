    libdis_test:      2f a4 04 10        lr.w s0,(s1)
    libdis_test+0x4:  af 24 09 14        lr.w.aq s1,(s2)
    libdis_test+0x8:  2f a9 09 12        lr.w.rl s2,(s3)
    libdis_test+0xc:  af 29 0a 16        lr.w.aqrl s3,(s4)
    libdis_test+0x10: 2f a4 24 19        sc.w s0,s2,(s1)
    libdis_test+0x14: af a4 34 1d        sc.w.aq s1,s3,(s1)
    libdis_test+0x18: 2f a9 44 1b        sc.w.rl s2,s4,(s1)
    libdis_test+0x1c: af a9 54 1f        sc.w.aqrl s3,s5,(s1)
    libdis_test+0x20: af a2 63 08        amoswap.w t0,t1,(t2)
    libdis_test+0x24: 2f 23 7e 0c        amoswap.w.aq t1,t2,(t3)
    libdis_test+0x28: af a3 ce 0b        amoswap.w.rl t2,t3,(t4)
    libdis_test+0x2c: 2f 2e df 0f        amoswap.w.aqrl t3,t4,(t5)
    libdis_test+0x30: af a2 63 00        amoadd.w t0,t1,(t2)
    libdis_test+0x34: 2f 23 7e 04        amoadd.w.aq t1,t2,(t3)
    libdis_test+0x38: af a3 ce 03        amoadd.w.rl t2,t3,(t4)
    libdis_test+0x3c: 2f 2e df 07        amoadd.w.aqrl t3,t4,(t5)
    libdis_test+0x40: af a2 63 20        amoxor.w t0,t1,(t2)
    libdis_test+0x44: 2f 23 7e 24        amoxor.w.aq t1,t2,(t3)
    libdis_test+0x48: af a3 ce 23        amoxor.w.rl t2,t3,(t4)
    libdis_test+0x4c: 2f 2e df 27        amoxor.w.aqrl t3,t4,(t5)
    libdis_test+0x50: af a2 63 60        amoand.w t0,t1,(t2)
    libdis_test+0x54: 2f 23 7e 64        amoand.w.aq t1,t2,(t3)
    libdis_test+0x58: af a3 ce 63        amoand.w.rl t2,t3,(t4)
    libdis_test+0x5c: 2f 2e df 67        amoand.w.aqrl t3,t4,(t5)
    libdis_test+0x60: af a2 63 40        amoor.w t0,t1,(t2)
    libdis_test+0x64: 2f 23 7e 44        amoor.w.aq t1,t2,(t3)
    libdis_test+0x68: af a3 ce 43        amoor.w.rl t2,t3,(t4)
    libdis_test+0x6c: 2f 2e df 47        amoor.w.aqrl t3,t4,(t5)
    libdis_test+0x70: af a2 63 80        amomin.w t0,t1,(t2)
    libdis_test+0x74: 2f 23 7e 84        amomin.w.aq t1,t2,(t3)
    libdis_test+0x78: af a3 ce 83        amomin.w.rl t2,t3,(t4)
    libdis_test+0x7c: 2f 2e df 87        amomin.w.aqrl t3,t4,(t5)
    libdis_test+0x80: af a2 63 a0        amomax.w t0,t1,(t2)
    libdis_test+0x84: 2f 23 7e a4        amomax.w.aq t1,t2,(t3)
    libdis_test+0x88: af a3 ce a3        amomax.w.rl t2,t3,(t4)
    libdis_test+0x8c: 2f 2e df a7        amomax.w.aqrl t3,t4,(t5)
    libdis_test+0x90: af a2 63 c0        amominu.w t0,t1,(t2)
    libdis_test+0x94: 2f 23 7e c4        amominu.w.aq t1,t2,(t3)
    libdis_test+0x98: af a3 ce c3        amominu.w.rl t2,t3,(t4)
    libdis_test+0x9c: 2f 2e df c7        amominu.w.aqrl t3,t4,(t5)
    libdis_test+0xa0: af a2 63 e0        amomaxu.w t0,t1,(t2)
    libdis_test+0xa4: 2f 23 7e e4        amomaxu.w.aq t1,t2,(t3)
    libdis_test+0xa8: af a3 ce e3        amomaxu.w.rl t2,t3,(t4)
    libdis_test+0xac: 2f 2e df e7        amomaxu.w.aqrl t3,t4,(t5)
